subprojects/language-java/src/integTest/groovy/org/gradle/language/java/JavaLanguageIncrementalBuildIntegrationTest.groovy

```diff
@@ -38,4 +38,33 @@ class JavaLanguageIncrementalBuildIntegrationTest extends AbstractJvmLanguageInc
         executedAndNotSkipped ":compileMainJarMainJava", ":createMainJar", ":mainJar"
     }
 
-}
\ No newline at end of file
+    def "task outcome is up to date when no recompilation necessary"() {
+        given:
+        buildFile << '''
+            tasks.withType(org.gradle.api.tasks.compile.AbstractCompile) { task ->
+                task.options.incremental = true
+                task.inputs.file 'input.txt' 
```

Comment:
I still feel pretty strong about not using the `task.inputs` api to trigger incremental compilation which does no work. I really think we should go down the multi project road with a use case where the incremental compilation would do no work without adding artifical inputs. Even better, then we can see when recompilation should not happen. Don't we have a test for that already?